serde/registry/example_test.go

```diff
@@ -0,0 +1,80 @@
+package registry_test
```

Comment:
I had the hope it would make things clearer but honestly I still struggle. example_test.go and registry_test.go seems to do the same, but with slightly different means, and this is not clear what the differences are. I'm still lost.

Reply:
As discussed, I merged the examples and will do a following PR to add another format to demonstrate the use case.